Tasks/AzureResourceGroupDeploymentV2/models/DeployAzureRG.ts

```diff
@@ -122,7 +122,7 @@ export class AzureRGTaskParameters {
             this.deploymentGroupName = tl.getInput("deploymentGroupName");
             this.copyAzureVMTags = tl.getBoolInput("copyAzureVMTags");
             var deploymentGroupEndpointName = tl.getInput("deploymentGroupEndpoint", false);
-            if(deploymentGroupEndpointName){
+            if(this.enableDeploymentPrerequisites && this.enableDeploymentPrerequisites == "ConfigureVMWithDGAgent"){
```

Comment:
do you need the 1st condition at all. Wont the equality check be sufficient

Reply:
Also do we need to handle casing. Just see how are we using it at other places

Reply:
1st condition check is just for safety, so that tomorrow if some visibility rule gets added to the input, it won't break.

Casing check won't be required, since it is a task input value(and hence fixed). At other places also, casing check is not done.

Reply:
ok